src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -538,6 +544,60 @@ func (c controller) addRouterCAToClusterCA() bool {
 
 }
 
+// TODO: remove
+type OLMOperator struct {
+	name      string
+	namespace string
+	enabled   bool
+	status    string
+}
+
+// TODO: remove
+var olms = map[string]OLMOperator{
+	"lso": {
+		"local-storage-operator.4.6.0-202101300140.p0", "openshift-local-storage", true, "",
+	},
+}
+
+func (c controller) getOLMOperators() ([]OLMOperator, error) {
+	ret := []OLMOperator{}
+	operators, err := c.ic.GetClusterOLM(context.TODO())
+	if err != nil {
+		c.log.WithError(err).Warningf("Failed to connect to assisted service")
+		return ret, err
+	}
+	for _, operator := range *operators {
+		if *operator.Enabled && operator.Status != "Succeeded" {
+			ret = append(ret, olms[string(operator.OperatorType)])
+		}
+	}
+	return ret, nil
+}
+
+// TODO: remove
+func updateStatus(olm OLMOperator, phase string) {
+	olm.status = phase
+	olms[olm.name] = olm
+}
+
+func (c controller) waitForOLMOperators() bool {
+	c.log.Infof("Checking OLM operators")
+	opeartors, _ := c.getOLMOperators()
+	if len(opeartors) == 0 {
```

Comment:
Would be good to add operator status based exit criteria so we won't wait till the timeout occurs.

Reply:
Well, it is. The `getOLMOperators` returns only operattors which are `enabled` and are not in `Succeeded` state. So we monitor only CSVs which are not in `Succeeded` state. But we could actually improve to also check for failures, and don't monitor the CSV which end in failed state.

Reply:
s/opeartors/operators